src/plonk/mod.rs

```diff
@@ -571,14 +613,30 @@ impl<F: PrimeField> TableData<F> {
         self.fixed = vec![vec![F::ZERO.into(); n]; self.cs.num_fixed_columns()];
         self.selector = vec![vec![false; n]; self.cs.num_selectors()];
         self.advice = vec![vec![F::ZERO.into(); n]; self.cs.num_advice_columns()];
+        config
+    }
+
+    // TODO Change design
+    pub(crate) fn postpone_assembly(&mut self) {
```

Comment:
what's the purpose to split fn assembly into 3 small functions?

Reply:
I just haven't chosen the final design for using TableData in IVC yet, but it definitely won't be an `assembly` function, so I've summarised all the code that isn't about `halo2::Circuit`